### PR TITLE
feat: handle fallback hex collor error

### LINKED ISSUE #72: Android always gives unknown color error when fallback color is #000
# Bug

When I call `getColors` using the following example code.

```
import React from 'react'
import { getColors } from 'react-native-image-colors'

const useImageColors = () => {
  const [colors, setColors] = React.useState(null)

  React.useEffect(() => {
    const url = 'https://i.imgur.com/68jyjZT.jpg'

    getColors(url, {
      fallback: '#000',
      cache: true,
      key: url,
    }).then(setColors)
  }, [])

  return colors
}
```

I keep getting the following error "Unknown color":

Screenshot 2023-06-29 at 3 17 09 PM

Notice the #000 as the fallback. 

## Environment info

| Library                   | Version |
| ------------------------- | ------- |
| react-native-image-colors | 2.1.0  |

I'm using a Samsung Galaxy Tab S6 Lite (New - 2023) 

## Steps To Reproduce

1. Use an Android device 
2. Use the example code above on version 2.1.0
3. Make sure the fallback is set to `#000`, this is very important to reproduce the bug
4. Check the console for the error "Unknown color"

Describe what you expected to happen:

1. I expected it to detect the color like it does for iOS. This `#000` fallback works fine for iOS but not for Android.
2. I also think the error message could be improved it should say `Unknown fallback color` so its clear why this error is occurring. It took me awhile to figure this out.

## Reproducible sample code

```
import React from 'react'
import { getColors } from 'react-native-image-colors'

const useImageColors = () => {
  const [colors, setColors] = React.useState(null)

  React.useEffect(() => {
    const url = 'https://i.imgur.com/68jyjZT.jpg'

    getColors(url, {
      fallback: '#000', // <---------------- IMPORTANT for bug to reproduce on Android
      cache: true,
      key: url,
    }).then(setColors)
  }, [])

  return colors
}
```

### PR DESCRIPTION
**Pull request by KrzysztofMoch**
# Summary
Add handler for fallback hex collor error. Also add android support for short hex codes (e.g `#000`)

closes #72 

# Test Plan
- [x] Build and run example
- [x] Change fallback color to:
  -  `#000` - should work on android and ios
  -  `#0000` - should throw error with message
  -  `#000000` - should work on android and ios

### CODE REVIEW

File: android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt [hunk 1]
```diff
@@ -78,6 +78,18 @@ class ImageColorsModule : Module() {
...
+  private fun verifyFallbackColor(hex: String): String {
+    if(!hex.matches(Regex("^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$"))) {
+      throw Exception("Invalid fallback hex color. Must be in the format #ffffff or #fff")
+    }
+
+    if(hex.length == 7) {
+      return hex
+    }
+
+    return "#${hex[1]}${hex[1]}${hex[2]}${hex[2]}${hex[3]}${hex[3]}"
```
Comment (watadarkstar) [inline android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt:90]: @KrzysztofMoch  I'm new to Kotlin. Does Kotlin start their index at 1 not 0? In JavaScript this code would look like:

```
    return "#${hex[0]}${hex[0]}${hex[1]}${hex[1]}${hex[2]}${hex[2]}"
```

File: android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt [hunk 2]
```diff
@@ -78,6 +78,18 @@ class ImageColorsModule : Module() {
     }
   }
 
+  private fun verifyFallbackColor(hex: String): String {
```
Comment (watadarkstar) [inline android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt:81]:
```suggestion
  private fun parseFallbackColor(hex: String): String {
```

File: android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt [hunk 3]
```diff
@@ -92,7 +104,8 @@ class ImageColorsModule : Module() {
     AsyncFunction("getColors") { uri: String, config: Config, promise: Promise ->
       service.launch {
         try {
-          val fallbackColorInt = Color.parseColor(config.fallback)
+          val fallbackColor = verifyFallbackColor(config.fallback)
```
Comment (watadarkstar) [inline android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt:107]:
```suggestion
          val fallbackColor = parseFallbackColor(config.fallback)
```

File: ios/ImageColorsModule.swift [hunk 4]
```diff
@@ -44,6 +45,27 @@ public class ImageColorsModule: Module {
         return String(format: "#%06X", rgb)
     }
 
+    private func verifyFallbackColor(hexColor: String) -> String? {
```
Comment (watadarkstar) [inline ios/ImageColorsModule.swift:48]:
```suggestion
    private func parseFallbackColor(hexColor: String) -> String? {
```

File: ios/ImageColorsModule.swift [hunk 5]
```diff
@@ -59,7 +81,13 @@ public class ImageColorsModule: Module {
         Name("ImageColors")
 
         AsyncFunction("getColors") { (uri: String, config: Config, promise: Promise) in
-            let fallbackColor = config.fallback
+            guard let fallbackColor = verifyFallbackColor(hexColor: config.fallback) else {
```
Comment (watadarkstar) [inline ios/ImageColorsModule.swift:84]:
```suggestion
            guard let fallbackColor = parseFallbackColor(hexColor: config.fallback) else {
```

**Review by watadarkstar — COMMENTED** [review]
This looks great. Thank you for doing this so quickly!

I left some minor suggestions and a question. :)

File: android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt (hunk 1 above)
Comment (KrzysztofMoch) [reply android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt:90]: Kotlin start at index 0, but we have string like `#000` so index 0 is `#`

**Review by osamaqarem — COMMENTED** [review]
Great job! There's just one thing to adjust.

File: android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt (hunk 1 above)
Comment (watadarkstar) [reply android/src/main/java/com/reactnativeimagecolors/ImageColorsModule.kt:90]: Oh right! Makes sense!

File: ios/ImageColorsModule.swift [hunk 6]
```diff
@@ -59,7 +81,13 @@ public class ImageColorsModule: Module {
...
+                let error = NSError.init(domain: ImageColorsModule.ERRORS.INVALID_FALLBACK_COLOR, code: -1)
+                DispatchQueue.main.asyncAfter(deadline: .now() + 3.0) {
+                    promise.reject(error)
+                }
```
[the remark this answers is not in the document]
Comment (KrzysztofMoch) [reply ios/ImageColorsModule.swift]: done ✅

**Review by osamaqarem — APPROVED** [review]

**Pull request merged**